### DISCUSSION TITLE
Help with organising data

### DISCUSSION THREAD

**Discussion by MarkJThomas** [bmcclintock/momentuHMM#134] [Q&A]
I hope it is okay to post here @TheoMichelot @bmcclintock, if there is a different forum please let me know.

I have been trying to run the HMM on my data but have come across a few things I couldn't find answers to online. Please see info and plots below.

My data: Vultures, GPS tags collecting hourly points from 5am to 7pm, daily for 8 months.

Here is a histogram of the time diff after I have thinned to one hour using the move package:

![time_intervals](https://user-images.githubusercontent.com/41441629/163022467-16ff102d-b703-43b0-8532-ce6165ea0fe6.jpeg)

Here is the same histogram after adding in the NAs:

![time_interval_60](https://user-images.githubusercontent.com/41441629/163022463-2a4c6110-d435-4cb5-8509-6cfd0371d362.jpeg)

**- Should I pad with NAs from 7pm to 5am? Or is everyday a new track?** 

When selecting initial parameter values, the turning angles histogram is Uniform (see plot) - how do I determine initial parameter values from this?

![hist_angle](https://user-images.githubusercontent.com/41441629/163022447-f11302ff-0b29-42ac-80bf-5a55830b4d8b.jpeg)

I struggled to find initial step parameters as well from the data given the step hist is from 0 - 30km 

![hist_step](https://user-images.githubusercontent.com/41441629/163022457-8e63d4a2-351e-49c7-9a5c-1a7dcd56640c.jpeg)

I've uploaded the CSV in case it helps but if anyone has any guidance or help I would be greatly appreciated!

[data_na.csv](https://github.com/bmcclintock/momentuHMM/files/8475387/data_na.csv)

I tried using random starting points but the code wouldn't work and I couldn't get it to work.

**Comment by bmcclintock** [comment]
It's difficult to provide any concrete advice without a reproducible example, but based on your description:

1) In `momentuHMM` there is no "correct" way to deal with the missing observations between 7pm and 5am. The options are to set them to NA, break up the tracks into segments without missing observations, or use multiple imputation to attempt to account for the missing observations. Given your data span 8 months, the best option(s) likely depends on what your study objectives are (e.g. short-term or long-term movement behaviors) and how much you know about vulture behavior during the periods of missing observations. For example, if vultures tend to be inactive at night, you might be able to get away with setting the missing observations to NA and maybe fixing these to the "inactive" state using the `knownStates` argument (and/or using a periodic function on the state transition probabilities as in the elephant example of the `momentuHMM` [vignette](https://cran.r-project.org/web/packages/momentuHMM/vignettes/momentuHMM.pdf)). As another example, if you are mostly interested in longer-term movement behavior, you could consider whittling the dataset to, say, only 1 observation per day.

2) For advice on starting values, see this moveHMM [vignette](https://cran.r-project.org/web/packages/moveHMM/vignettes/moveHMM-starting-values.pdf), #55, #84, and search for "initial values" in the GitHub [Discussions](https://github.com/bmcclintock/momentuHMM/discussions).

**Reply by MarkJThomas** [reply]
thank you @bmcclintock and apologies for the late response!

The missing observations are largely evening when the vulture's are inactive - I set them to NA, but I couldn't find how to do it by knownStates? I added a column (knownStates) to my csv and added 1 where it was inactive - but the other state is active (flying) which we want to draw out of the model. However, a lot of the behaviour between 7am - 6pm will be either resting, or foraging (flying). If setting the NAs to inactive, will the model still denote the two states from the daytime locations?

Thanks!

**Reply by bmcclintock** [reply]
> The missing observations are largely evening when the vulture's are inactive - I set them to NA, but I couldn't find how to do it by knownStates?

See the `knownStates` argument in `?fitHMM` and the examples using `knownStates` in the package [vignette](https://cran.r-project.org/web/packages/momentuHMM/vignettes/momentuHMM.pdf) and [example code](https://github.com/bmcclintock/momentuHMM/tree/master/vignettes/examples) (e.g. grey seal example). See also #68 and search for "knownStates" in the [GitHub Discussions](https://github.com/bmcclintock/momentuHMM/discussions). 

> If setting the NAs to inactive, will the model still denote the two states from the daytime locations?

I don't know enough about your system and data to answer this. If I'm understanding correctly, you should only set `knownStates` for the NAs that you're willing to assume were indeed inactive (i.e. during nighttime). To distinguish the two states during daytime, there would need to be at least some observed periods of inactivity during the daytime.
